frontend/src/metabase/archive/containers/ArchiveApp.tsx

```diff
@@ -0,0 +1,182 @@
+
+interface ArchiveAppRootProps {
+  list: EntityWrappedCollectionItem[];
+  reload: () => void;
+}
+
+function ArchiveAppRoot({ list, reload }: ArchiveAppRootProps) {
+  const mainElement = useMemo(() => getMainElement(), []);
+  useEffect(() => {
+    if (!isSmallScreen()) {
+      openNavbar();
+    }
+  }, []);
+
+  const { clear, getIsSelected, selected, selectOnlyTheseItems, toggleItem } =
+    useListSelect(item => `${item.model}:${item.id}`);
+
+  const selectAllItems = useCallback(() => {
+    selectOnlyTheseItems(list);
+  }, [list, selectOnlyTheseItems]);
+
+  const allSelected = useMemo(
+    () => selected.length === list.length,
+    [selected, list],
+  );
+
+  const isNavbarOpen = useSelector(getIsNavbarOpen);
+  const isAdmin = useSelector(getUserIsAdmin);
+
+  return (
+    <ArchiveRoot>
+      <ArchiveHeader>
+        <PageHeading>{t`Archive`}</PageHeading>
+      </ArchiveHeader>
+      <ArchiveBody>
+        <Card
+          style={{
+            height: list.length > 0 ? ROW_HEIGHT * list.length : "auto",
+          }}
+        >
+          {list.length > 0 ? (
+            <VirtualizedList
+              scrollElement={mainElement}
+              items={list}
+              rowHeight={ROW_HEIGHT}
+              renderItem={({ item }: { item: EntityWrappedCollectionItem }) => (
+                <ArchivedItem
+                  type={item.type || ""}
+                  name={item.getName()}
+                  icon={item.getIcon().name}
+                  color={item.getColor()}
+                  isAdmin={isAdmin}
+                  onUnarchive={async () => {
+                    if (item.setArchived !== undefined) {
+                      await item.setArchived(false);
+                      reload();
+                    }
+                  }}
+                  onDelete={async () => {
+                    if (item.delete !== undefined) {
+                      await item.delete();
+                      reload();
+                    }
+                  }}
+                  selected={getIsSelected(item)}
+                  onToggleSelected={() => toggleItem(item)}
+                  showSelect={selected.length > 0}
+                />
+              )}
+            />
+          ) : (
+            <ArchiveEmptyState>
+              <h2>{t`Items you archive will appear here.`}</h2>
+            </ArchiveEmptyState>
+          )}
+        </Card>
+      </ArchiveBody>
+      <BulkActionBar isNavbarOpen={isNavbarOpen} showing={selected.length > 0}>
+        <ArchiveBarContent>
+          <SelectionControls
+            allSelected={allSelected}
+            selectAll={selectAllItems}
+            clear={clear}
+          />
+          <BulkActionControls selected={selected} reload={reload} />
+          <ArchiveBarText>{t`${selected.length} items selected`}</ArchiveBarText>
+        </ArchiveBarContent>
+      </BulkActionBar>
+    </ArchiveRoot>
+  );
+}
+
+export const ArchiveApp = _.compose(
+  Search.loadList({
```

Comment:
Can we leverage the `use-search-list-query` hook rather than the entity loader HoC?

Reply:
Haha, I see you actually referenced this in the PR Description. Is there any particular reason we should keep Seach.loadList?

Reply:
The reason I kept it is because it seemed like the hardest refactor to do since `use-search-list-query` would require some modifications before it could be a drop-in replacement. Currently `Search.loadList` uses a `wrapped: true` option that affects the behavior of the list and `use-search-list-query` does not support that (to my knowledge).

I will investigate it more to see what changes would need to be made to make `use-search-list-query` viable here.